File: packages/android/src/main/java/com/kozmos/components/MapControlsGroup/MapControlsGroup.figma.kt
package com.kozmos.components.mapcontrolsgroup

import androidx.compose.runtime.Composable
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType
import com.kozmos.components.mapcontrolbutton.KozmosMapControlButtonPresentation

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=1351-8408")
class KozmosMapControlsGroupConnect {
    @FigmaProperty(FigmaType.Text, "Location Label Text")
    val locationLabel: String = "My location"

    @FigmaProperty(FigmaType.Enum, "LocationPresentation")
    val locationPresentation: KozmosMapControlButtonPresentation = Figma.mapping(
        "IconOnly" to KozmosMapControlButtonPresentation.IconOnly,
        "Labelled" to KozmosMapControlButtonPresentation.Labelled
    )

    @Composable
    fun ComponentExample() {
        KozmosMapControlsGroup(
            compassBearing = 0f,
            onZoomIn = {},
            onZoomOut = {},
            onCompassReset = {},
            onMyLocation = {},
            locationPresentation = locationPresentation,
            locationLabel = locationLabel
        )
    }
}

File: packages/ios/Sources/Components/MapControlsGroup/MapControlsGroup.figma.swift
import SwiftUI
import Figma

struct KozmosMapControlsGroupConnect: FigmaConnect {
    let component = KozmosMapControlsGroup.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=1351-8408"

    @FigmaString("Location Label Text")
    var locationLabel: String = "My location"

    @FigmaEnum(
        "LocationPresentation",
        mapping: [
            "IconOnly": KozmosMapControlButtonPresentation.iconOnly,
            "Labelled": KozmosMapControlButtonPresentation.labelled
        ]
    )
    var locationPresentation: KozmosMapControlButtonPresentation = .iconOnly

    var body: some View {
        KozmosMapControlsGroup(
            compassBearing: 0,
            onZoomIn: {},
            onZoomOut: {},
            onCompassReset: {},
            onMyLocation: {},
            locationPresentation: self.locationPresentation,
            locationLabel: self.locationLabel
        )
    }
}
